data-prepper-plugins/saas-source-plugins/jira-source/src/main/java/org/opensearch/dataprepper/plugins/source/jira/rest/BasicAuthInterceptor.java

```diff
@@ -0,0 +1,32 @@
+package org.opensearch.dataprepper.plugins.source.jira.rest;
+
+import org.opensearch.dataprepper.plugins.source.jira.JiraSourceConfig;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+
+public class BasicAuthInterceptor implements ClientHttpRequestInterceptor {
```

Comment:
This should probably be in the Crawler package since it is a standard. Also, we may have some existing code for this to re-use. Please check.

Reply:
Each connector may have different ways to authorizing a request. This will be common between all of Atlassian services but may not be common for other connectors like Salesforce or Service now etc

Reply:
Basic authentication is common though.

But, we can address this later.